packages/terra-functional-testing/src/services/wdio-visual-regression-service/methods/BaseCompare.js

```diff
@@ -1,27 +1,128 @@
 /* eslint-disable class-methods-use-this, no-unused-vars */
+import path from 'path';
+
+
+const TEST_ID_REGEX = /\[([^)]+)\]/;
 
 export default class BaseCompare {
+  constructor(options) {
+    const {
+      baseScreenshotDir,
+      locale,
+      theme,
+    } = options;
+
+    // screenshot naming config
+    this.baseScreenshotDir = baseScreenshotDir || process.cwd();
+    this.locale = locale || 'en';
+    this.theme = theme || 'terra-default-theme';
+  }
+
   /**
    * You can do here your image comparison magic.
+   * @param {Object} context - information provided to process the screenshot
+   * @param {Object} context.browser - { name, version, userAgent }
+   * @param {Object} context.suite - the test suite that is running
+   * @param {Object} context.test - the test that is running
+   * @param {Object} context.meta - { element, exclude, hide, remove, viewport}
+   * @param {*} base64Screenshot - the screenshot captured by the selenium command to process.
    */
-  async processScreenshot(context, base64Screenshot) {
+  async processScreenshot(_context, _base64Screenshot) {
```

Comment:
What does defining this function do and what happens if it's removed?  Is this class used as an interface for subclassing? Is it required to be async?

Reply:
This function is expected to be overridden and provided by the sub class that extends it. This is async to allow image comparison to finish before moving on to the next command. I'll update the description.